src/coreclr/build.cmd

```diff
@@ -299,6 +297,8 @@ if "%__NMakeMakefiles%"=="1" (set "__IntermediatesDir=%__RootBinDir%\nmakeobj\%_
 set "__PackagesBinDir=%__BinDir%\.nuget"
 set "__CrossComponentBinDir=%__BinDir%"
 set "__CrossCompIntermediatesDir=%__IntermediatesDir%\crossgen"
+set "__LogsDir=%__RootBinDir%\log\!__BuildType!"
+set "__MsbuildDebugLogsDir=%__LogsDir%\MsbuildDebugLogs"
```

Comment:
Curious: do Mono builds have the same problem as you're fixing here?

Reply:
I can't speak for the state of all of their infrastructure, given I don't know it at all. However, I see *some* binlogs getting published in the official build integration work that @directhex has here: https://dev.azure.com/dnceng/internal/_build/results?buildId=537708&view=artifacts&type=publishedArtifacts
And it's exactly what arcade does. 

These are, however, only for the Pack target. I would rather open an issue and deal with it in another pass. I wanted to do this before trying to fix the change that was reverted.